### PR TITLE
Avoid C4703 error on UWP

### PR DESCRIPTION
**Pull request by esteve**
MSVC with stricter flags complains that some variables might be used uninitialized. See:

https://ci.appveyor.com/project/esteve/ros2-dotnet/build/1.0.70/job/g7nnyq1huwyml1hp

https://msdn.microsoft.com/en-us/library/jj851030.aspx?f=255&MSPPError=-2147217396

Connect to ros2/rcl#280

### CODE REVIEW

**Comment by dirk-thomas** [comment]
Since `res` is not set anywhere I would suggest the following instead:

* keep the `return RCL_RET_ERROR` as it was 
* remove the condition `if (RCL_RET_OK == res) {`
* replace the final return with `return RCL_RET_OK`

This gets rid of `res` altogether.

File: rcl_yaml_param_parser/src/parser.c [hunk 1]
```diff
@@ -169,8 +169,7 @@ static rcl_ret_t add_name_to_ns(
       sep_str = PARAMETER_NS_SEPERATOR;
       break;
     default:
-      res = RCL_RET_ERROR;
-      break;
+      return RCL_RET_ERROR;
```
Comment (wjwwood) [inline rcl_yaml_param_parser/src/parser.c]: Could you also remove the following `if (RCL_RET_OK == res)` condition and the `res` variable all together? I'd just do it myself, but I wanted to see what you think about it.

File: rcl_yaml_param_parser/src/parser.c [hunk 2]
```diff
@@ -240,8 +239,7 @@ static rcl_ret_t rem_name_from_ns(
...
```
Comment (wjwwood) [inline rcl_yaml_param_parser/src/parser.c]: Same here.

**Comment by dirk-thomas** [comment]
@esteve Friendly ping.

**Comment by esteve** [comment]
@dirk-thomas thanks for the reminder, sorry for the late response.. I've addressed the style issues, do the changes look the way you and @wjwwood described?

**Comment by dirk-thomas** [comment]
Currently CI is failing:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/5299/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/2053/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/4416/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/5279/)

@esteve Can you please rebase this branch to the latest state from master.

**Comment by dirk-thomas** [comment]
@esteve Friendly ping.

**Comment by nuclearsandwich** [comment]
@esteve does this build error still occur with the changes in rcl since this PR was opened? Are you able to update it or should we close it out?

**Comment by esteve** [comment]
@nuclearsandwich let me rebase this and check if it still is necessary.

**Comment by esteve** [comment]
@nuclearsandwich I've rebased this PR and reapplied the feedback from @dirk-thomas and @wjwwood

File: rcl_yaml_param_parser/src/parser.c [hunk 3]
```diff
@@ -181,54 +180,51 @@ static rcutils_ret_t add_name_to_ns(
...
-    if (NULL == name) {
-      return RCUTILS_RET_INVALID_ARGUMENT;
+  /// Add a name to ns
+  if (NULL == name) {
+    return RCUTILS_RET_INVALID_ARGUMENT;
+  }
+  if (0U == *cur_count) {
+    cur_ns = rcutils_strdup(name, allocator);
+    if (NULL == cur_ns) {
+      return RCUTILS_RET_BAD_ALLOC;
```
Comment (dirk-thomas) [inline rcl_yaml_param_parser/src/parser.c]: Isn't `cur_ns` leaked in this case?

Comment (dirk-thomas) [reply rcl_yaml_param_parser/src/parser.c]: Nvm - it is NULL.

**Comment by dirk-thomas** [comment]
CI builds for `rcl_yaml_param_parser`:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/7773/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/3863/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/6352/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/7615/)

**Comment by dirk-thomas** [comment]
Thanks for the patch.

**Comment by theseankelly** [comment]
Hey @dirk-thomas can this be considered for backport to Dashing?

**Comment by nuclearsandwich** [comment]
> can this be considered for backport to Dashing?

In order to be backported this change must not introduce any changes to the Dashing ABI. I haven't looked closely at it but if it meets that criteria you can open a PR against the `dashing` branch of this repository cherry picking this change and mention me on it. I'll take care of updating the Dashing patch release project board, which is now tracking this RP

**Pull request merged**